protocols/shell-foreign-toplevel-geometry-tracker-v1.xml

```diff
@@ -0,0 +1,267 @@
+    SPECIAL, INDIRECT OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN
+    AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION,
+    ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF
+    THIS SOFTWARE.
+  </copyright>
+
+  <description summary="protocol allowing clients to track foreign toplevel geometry">
+    This protocol allows clients to track the geometry of toplevels
+    relative to outputs or workspaces, enabling workspace overviews
+    and window picking.
+
+    This protocol is privileged and should be limited to trusted clients.
+
+    The key words "must", "must not", "required", "shall", "shall not",
+    "should", "should not", "recommended",  "may", and "optional" in this
+    document are to be interpreted as described in IETF RFC 2119.
+
+    Warning! The protocol described in this file is intended as a stopgap
+    and is expected to be superseded by a solution in wayland-protocols.
+    Clients should not assume this protocol will continue to exist in the
+    future.
+  </description>
+
+  <interface name="shell_foreign_toplevel_geometry_tracker_manager_v1" version="1">
+    <description summary="manager for geometry trackers">
+      This global creates and synchronizes geometry trackers.
+    </description>
+
+    <enum name="error">
+      <entry name="live_trackers"
+             summary="destroyed geometry tracker manager while trackers were live"
+             value="1"/>
+    </enum>
+      
+    <request name="destroy" type="destructor">
+      <description summary="destroy the manager">
+        Destroy the manager. Destroying the manager prior to
+        the destruction of all trackers created by it is a protocol
+        error, as the manager is responsible for synchronization.
+      </description>
+    </request>
+
+    <event name="done">
+      <description summary="all geometry tracker events have been sent">
+        This event signals that all events related to created geometry
+        trackers and their members have been sent.
+      </description>
+    </event>
+
+    <request name="get_output_tracker">
+      <description summary="create a geometry tracker for a given output">
+        Creates a geometry tracker which tracks toplevels on a given output.
+      </description>
+
+      <arg name="tracker" type="new_id" interface="shell_foreign_toplevel_geometry_tracker_v1"/>
+      <arg name="output" type="object" interface="wl_output"/>
+    </request>
+
+    <request name="get_workspace_tracker">
+      <description summary="create a geometry tracker for a given workspace">
+        Creates a geometry tracker which tracks toplevels on a given workspace.
+      </description>
+
+      <arg name="tracker" type="new_id" interface="shell_foreign_toplevel_geometry_tracker_v1"/>
+      <arg name="workspace" type="object" interface="ext_workspace_handle_v1"/>
+    </request>
```

Comment:
@vaxerski thoughts on splitting this apart similarly to the image capture protocol? I'm not sure what sources other than a workspace or output would make sense but it might make sense to at least make workspace support optional.

Reply:
meh, I don't see any other possible tracker, so I don't really care